### ISSUE TITLE
Does not work properly using iso8601 date formats in timezones

### ISSUE THREAD

**Issue by bterkuile** [fomantic/Fomantic-UI#1495]
When using the calendar for date picking using iso8601 strings (2020-05-12) the display date changes due to the use of the internal javascript date parser:

https://github.com/fomantic/Fomantic-UI/blob/3f3045c09176875bd3db674c0874bc4b96d0d3f7/src/definitions/modules/calendar.js#L1346

This has issues well described at: https://stackoverflow.com/questions/7556591/is-the-javascript-date-object-always-one-day-off

To avoid native javascript Date 'wonkeyness' I personally use the `moment` library. Since that is not a desirable dependency here I would not an easy solution.

To reproduce (changin the day of the original iso8601 inputs (negative time zones) (2020-05-12 => 2020-05-11)):
```javascript
$('.datepicker').calendar({
  firstDayOfWeek: 1,
  type: 'date',
  formatter: {
    datetime: function(date, settings) {
      if (!date) {
        return '';
      }
      return moment(date).format().substr(0, 10);
    }
  }
});
```
when correcting for this and switching back to utc on formatting, there can/will be a day change when selecting a date from the calendar (note the difference is the `utc` call):
```javascript
$('.datepicker').calendar({
  firstDayOfWeek: 1,
  type: 'date',
  formatter: {
    datetime: function(date, settings) {
      if (!date) {
        return '';
      }
      return moment(date).utc().format().substr(0, 10);
    }
  }
});
```

**Comment by lubber-de** [comment]
This should be fixed by #1492

**Comment by lubber-de** [comment]
Duplicate of #1462
